src/mono/netcore/System.Private.CoreLib/src/System/Threading/Thread.Mono.cs

```diff
@@ -254,6 +255,7 @@ public static void Sleep(int millisecondsTimeout)
         [UnsupportedOSPlatform("browser")]
```

Comment:
The CoreCLR and Mono implementations look very similar. Can we move some of this into shared CoreLib?

Reply:
I did a scan for what could be shared, it looks like there are lots of things that could be shared in the path to starting a thread in general, to me it a good refactor would be a rewrite of that path that is close to but different from coreclr implementation. That would be nice to do, but it doesn't seem to be trivial and probably is tangential from this change. There are maybe small things that could be shared as a result of this change, I don't see much benefit in that since most of the start paths are separated currently and it's a very small amount of code.

Reply:
Fair enough.

I may give this refactor a shot as I am porting this change to NativeAOT.